Sources/Main.cpp

```diff
@@ -839,10 +807,8 @@ void __fastcall TX584Form::CodeListViewDragOver(TObject *Sender,
             int Row = CodeListView->TopItem->Index + (Y - CodeListView->TopItem->Position.y) / LeftImageList->Height;
```

Comment:
I believe `Row` is calculated a bit incorrectly on HiDPI screens when scaling is used. Try to maximize the window and drag some microinstructions to the bottom of the code editor. The selected row will be misaligned with mouse cursor. This is better reproducible with higher scaling values, such as 150% or 200%.

This happens because `LeftImageList->Height` slighly differs from the true row height. On my machine `LeftImageList->Height` is 48 pixels while the row height is 47 pixels. There should be some way to calculate proper row height and use it instead of `LeftImageList->Height`. Can you please check the following line and see if it works for you:

```cpp
int RowHeight = CodeListView->TopItem->DisplayRect(drBounds).Height();
```

Reply:
1920x1200, 120dpi, bug has not been reproduced. But you are right